cypress-io/cypress
# fix: Extend mock logger used in angular bootstrap

- Closes #25312 

### User facing changelog
Fixes an issue where Angular Component Testing projects would fail to initialize if an unsupported `browserslist` entry was specified in the project configuration.

### Additional details
We were building an incomplete `logger` instance for use by `@angular/devkit` util functions - if an unsupported `browserslist` entry was specified the utils would try to make a logging call that would fail. This PR makes a couple changes:
1. Extend the logger to provide more of the expected contract. Full compliance felt like overkill (25+ fields, hence the `as unknown as logger.Logger` cast) but this should cover most use cases
2. Make the logger proxy output to `debug` so that it's available when troubleshooting - output would previously be swallowed
3. Minor performance tweak so an expected failure occurs faster in tests

### Steps to test
1. Create an Angular v15 project: `npx @angular/cli new my-project`
2. Create a `.browserslistrc` file with the content `IE 11`
3. Open project in CT using v12.3.0, observe failure: `npx cypress open --component`
4. Checkout this branch, do a `yarn build` in `npm/webpack-dev-server`, and open the project,. Observe no failure

### How has the user experience changed?
Project with unsupported `browserslist` config will now open

### PR Tasks
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
